src/main/java/edu/oregonstate/cs361/battleship/Main.java

```diff
@@ -362,26 +356,48 @@ private static BattleshipModelUpdated computerFireHard(BattleshipModelUpdated mo
             FireSpotComputer = new Point(shootX, shootY);
         } while ( alreadyShot( FireSpotComputer, model,false));
 
+        for(int i = 0; i < model.getPlayerHits().size(); i++){
+            Point checkPoint = model.getPlayerHits().get(i);
+
+            for (int k = 0; k < 5; k++) {
+                if (Hit(PArray[k].getStart(), PArray[k].getEnd(), checkPoint)) { //Finds the ship
+                    if(PArray[k].getHealth() > 0){ //Makes sure that the ship hasn't been sunk before (user knows this)
+
+                        Point Left = new Point(checkPoint.getAcross() - 1, checkPoint.getDown());
+                        Point Right = new Point(checkPoint.getAcross() + 1, checkPoint.getDown());
+                        Point Down = new Point(checkPoint.getAcross(), checkPoint.getDown() - 1);
+                        Point Up = new Point(checkPoint.getAcross(), checkPoint.getDown() + 1);
+
+                        if( !alreadyShot( Left, model,false) && Left.getAcross() > 0){
+                            FireSpotComputer = new Point(Left.getAcross(), Left.getDown());
+
+                        }else if( !alreadyShot( Right, model,false ) && Right.getAcross() < 11){
+                            FireSpotComputer = new Point( Right.getAcross(), Right.getDown());
+
+                        }else if( !alreadyShot( Down, model,false ) && Down.getDown() > 0){
+                            FireSpotComputer = new Point(Down.getAcross(), Down.getDown());
+
+                        }else if( !alreadyShot( Up, model,false ) && Up.getDown() < 11){
+                            FireSpotComputer = new Point(Up.getAcross(), Up.getDown());
+                        }
+                    }
+                }
+            }
+        }
+
```

Comment:
Can't wait to try it. Nice job naming of the variables it tells me where it is going to look at around the hit point